### PR TITLE
Apfelbaeumle

### PR DESCRIPTION
**Pull request by apfelbaeumle**
* small optimisation of database creation
* .eslintrc added

### CODE REVIEW

File: server/database_setup.sql [hunk 1]
```diff
@@ -1,14 +1,18 @@
 -- kybernetik-abrechnung requires postgres >= 10.5
 
-drop owned by abrechnung cascade;
+-- drop owned by abrechnung cascade;
```
Comment (mic-e) [inline server/database_setup.sql:3]: in `db.js` `async factory_reset()` runs this file as a query and relies on the fact that there is a drop statement on top.

I suggest to add the `drop` statement as an additional query in the `factory_reset` function, and completely delete it from here.

File: server/database_setup.sql [hunk 2]
```diff
@@ -88,25 +96,48 @@ create table group_memberships (
...
-  perform pg_notify('group_memberships', '');
-  return NEW;
+create index if not exists group_memberships_uid_index on group_memberships (uid);
+create index if not exists group_memberships_gid_index on group_memberships (gid);
+
+create or replace function group_memberships_notify_trigger() returns trigger as $$
+declare
+	lq bigint := nextval('last_mod_seq_counter');
+begin
+	with mod_uids AS (
```
Comment (mic-e) [inline server/database_setup.sql]: I think we want to delete the `mod_uids` part

File: server/database_setup.sql [hunk 3]
```diff
@@ -88,25 +96,48 @@ create table group_memberships (
...
+		select distinct id
+		from users, group_memberships
+		where id = uid and users.last_mod_seq < group_memberships.last_mod_seq
+	)
+	update users
```
Comment (mic-e) [inline server/database_setup.sql]: add a comment: what purpose does this serve? why do we need to update group?

File: server/database_setup.sql (hunk 1 above)
Comment (apfelbaeumle) [reply server/database_setup.sql:3]: done

File: server/database_setup.sql (hunk 2 above)
Comment (apfelbaeumle) [reply server/database_setup.sql]: completely removed as I dumped that idea

File: server/database_setup.sql (hunk 3 above)
Comment (apfelbaeumle) [reply server/database_setup.sql]: dumped that idea

**Pull request merged**